napari/viewer.py

```diff
@@ -143,6 +143,40 @@ def export_figure(
             flash=flash,
         )
 
+    def export_rois(
+        self,
+        rois: list[np.ndarray],
+        paths: Optional[list[str]] = None,
+        scale: Optional[float] = None,
+    ):
+        """Export the shapes rois with storage file paths
+
+        Parameters
+        ----------
+        rois: numpy array
+            An list of arrays with each having shape (4, 2) representing a rectangular roi. If not rectangular
+            a screenshot of the bounding box of the roi will be taken.
+        paths: list
+            The list to store file path for shapes roi
+
+        Returns
+        -------
+        roi_dict: dictionary
+            The dictionary with index and file paths for each shapes roi
+        """
+        # Check to see if roi has shape (n,2,2)
+        if any(roi.shape[-2:] != (4, 2) for roi in rois):
+            raise ValueError(
```

Comment:
Would be good to get coverage on this line by adding a test that passes e.g. a 3D roi.

Reply:
Ah yes, this makes sense and would add a test in next update.